src/network.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+
+import { EnforcementLevel, ResourceValidationPolicy, validateTypedResource } from "@pulumi/policy";
+
+import { registerPolicy } from "./awsGuard";
+import { defaultEnforcementLevel } from "./enforcementLevel";
+
+// Mixin additional properties onto AwsGuardArgs.
+declare module "./awsGuard" {
+    interface AwsGuardArgs {
+        albHttpToHttpsRedirection?: EnforcementLevel;
+    }
+}
+
+// Register policy factories.
+registerPolicy("albHttpToHttpsRedirection", albHttpToHttpsRedirection);
+
+/** @internal */
+export function albHttpToHttpsRedirection(enforcementLevel: EnforcementLevel): ResourceValidationPolicy {
```

Comment:
For now, I've been making the parameters of the factory function match the "signature" of the args, so making the enforcement level optional here.

```suggestion
export function albHttpToHttpsRedirection(enforcementLevel?: EnforcementLevel): ResourceValidationPolicy {
```